big.go

```diff
@@ -29,6 +29,26 @@ func NewBigIntFromInt64(v int64) *BigInt {
 	return &BigInt{new(big.Int).SetInt64(v)}
 }
 
+// NewBigIntFromString creates a BigInt by parsing a string.
+// nil will be returned if the given string is not parsable.
+func NewBigIntFromString(data string) *BigInt {
+	b, success := new(big.Int).SetString(data, 0)
+	if !success {
+		return nil
+	}
+	return &BigInt{b}
+}
+
+// NewBigIntFromStringBase creates a BigInt by parsing a string containing a number of given base.
+// nil will be returned if the given string is not parsable.
+func NewBigIntFromStringBase(data string, base int) *BigInt {
```

Comment:
Same here, return `(*BigInt, error)` here and if `!success`, return something like `errors.New("invalid number string")`.